src/main/webapp/auth-flow.js

```diff
@@ -0,0 +1,28 @@
+async function handleLogin() {
+  fetch('/status').then(response => response.json()).then((isLoggedIn) => {
+    if (!isLoggedIn) {
+      window.location.replace("/login");
+    }
+  });
+}
+
+function authorizeCallback(authResult) {
+  if (authResult['code']) {
+    $('#signinButton').attr('style', 'display: none');
+
+    const codeMsg = 'code=' + authResult['code'];
+    const request = new Request('/authorize', {
+                                                method: 'POST',
+                                                headers: {'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8', },
+                                                body: codeMsg});
+    fetch(request).then(response => {
+        console.log('auth code sent'); //TODO: handle failure case
```

Comment:
What is the failure case that we need to handle here?  Will this cause issues when users try to access the page in certain circumstances?

Reply:
The failure cases are (1) user has logged out (2) something happens when trying to get a refresh token from the google API.

Both cases are handled (on the server side, there will not be any bad state), but the user is not informed of it.

Reply:
Ah ok, I suppose for now this is good enough considering our top priority is to get the auth flow working, but we should try to alert the user of what the issue is at some point.